### ISSUE TITLE
Class FtpURLConnection is not accessible when using Java 9

### ISSUE THREAD

**Issue by baffelli** [nextflow-io/nextflow#646]
Using Java9 .Here is the full stack trace:
```
Apr-04 15:06:39.024 [Actor Thread 4] ERROR nextflow.processor.TaskProcessor - Execution aborted due to an unexpected error
java.lang.IllegalAccessError: class nextflow.file.http.XFileSystemProvider (in unnamed module @0x4450d156) cannot access class sun.net.www.protocol.ftp.FtpURLConnection (in module java.base) because module java.base does not export sun.net.www.protocol.ftp to unnamed module @0x4450d156
	at nextflow.file.http.XFileSystemProvider.readHttpAttributes(XFileSystemProvider.groovy:382)
	at nextflow.file.http.XFileSystemProvider.readAttributes(XFileSystemProvider.groovy:361)
	at java.base/java.nio.file.Files.readAttributes(Files.java:1755)
	at nextflow.util.CacheHelper.hashFile(CacheHelper.java:183)
	at nextflow.util.CacheHelper.hasher(CacheHelper.java:142)
	at nextflow.util.CacheHelper.hasher(CacheHelper.java:139)
	at nextflow.util.CacheHelper.hasher(CacheHelper.java:74)
	at nextflow.util.CacheHelper.hasher(CacheHelper.java:70)
	at nextflow.util.CacheHelper.hashUnorderedCollection(CacheHelper.java:256)
	at nextflow.util.CacheHelper.hasher(CacheHelper.java:130)
	at nextflow.util.CacheHelper.hasher(CacheHelper.java:134)
	at nextflow.util.CacheHelper.hasher(CacheHelper.java:74)
	at nextflow.util.CacheHelper.hasher(CacheHelper.java:70)
	at nextflow.util.CacheHelper$hasher.call(Unknown Source)
	at nextflow.processor.TaskProcessor.createTaskHashKey(TaskProcessor.groovy:1948)
	at jdk.internal.reflect.GeneratedMethodAccessor360.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:210)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:59)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:166)
	at nextflow.processor.TaskProcessor.invokeTask(TaskProcessor.groovy:676)
	at jdk.internal.reflect.GeneratedMethodAccessor253.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSite.invoke(PogoMetaMethodSite.java:169)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.call(PogoMetaMethodSite.java:71)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at nextflow.processor.TaskProcessor$InvokeTaskAdapter.call(TaskProcessor.groovy:224)
	at groovyx.gpars.dataflow.operator.DataflowOperatorActor.startTask(DataflowOperatorActor.java:120)
	at groovyx.gpars.dataflow.operator.ForkingDataflowOperatorActor.access$001(ForkingDataflowOperatorActor.java:35)
	at groovyx.gpars.dataflow.operator.ForkingDataflowOperatorActor$1.run(ForkingDataflowOperatorActor.java:58)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.base/java.lang.Thread.run(Thread.java:844)

```

**Comment by pditommaso** [comment]
Quite surely this is caused by Java 9/10 which is not yet fully supported. However I'm not understanding how this is happening. Can you also include the snippet (process) generating this issue ?

**Comment by pditommaso** [comment]
@baffelli Can you try to add the above entry just after [this line](https://github.com/nextflow-io/nextflow/blob/master/nextflow#L433) and verify if it solves the issue? 

```
launcher+=(--add-opens=java.base/jdk.internal.misc=ALL-UNNAMED)
```

**Comment by pditommaso** [comment]
Ok, I've managed to replicate it. The following option solve the problem `--add-exports=java.base/sun.net.www.protocol.ftp=ALL-UNNAMED`

**Comment by pditommaso** [comment]
Included in version `0.28.2`.
